Extensions/LinkedObjects/linkedobjects.ts

```diff
@@ -112,30 +216,125 @@ namespace gdjs {
         LinksManager.getManager(runtimeScene).removeAllLinksOf(objA);
       };
 
-      export const _objectIsInList = function (
-        obj: gdjs.RuntimeObject,
-        linkedObjects: gdjs.RuntimeObject[]
+      export const quickPickObjectsLinkedTo = function (
+        runtimeScene: gdjs.RuntimeScene,
+        eventsFunctionContext: EventsFunctionContext | undefined,
+        objectsLists: Hashtable<gdjs.RuntimeObject[]>,
+        obj: gdjs.RuntimeObject
       ) {
-        return linkedObjects.indexOf(obj) !== -1;
+        return doPickObjectsLinkedTo(
+          runtimeScene,
+          objectsLists,
+          obj,
+          !!eventsFunctionContext
+        );
       };
 
       export const pickObjectsLinkedTo = function (
         runtimeScene: gdjs.RuntimeScene,
         objectsLists: Hashtable<gdjs.RuntimeObject[]>,
         obj: gdjs.RuntimeObject
+      ) {
+        return doPickObjectsLinkedTo(
+          runtimeScene,
+          objectsLists,
+          obj,
+          // Can't know if it's called from an event function or not
+          // Object names will have to be checked
+          true
+        );
+      };
+
+      const doPickObjectsLinkedTo = function (
+        runtimeScene: gdjs.RuntimeScene,
+        objectsLists: Hashtable<gdjs.RuntimeObject[]>,
+        obj: gdjs.RuntimeObject,
+        isEventsFunction: boolean
       ) {
         if (obj === null) {
           return false;
         }
-        const linkedObjects = LinksManager.getManager(
+        const linkedObjectMap = LinksManager.getManager(
           runtimeScene
-        ).getObjectsLinkedWith(obj);
-        return gdjs.evtTools.object.pickObjectsIf(
-          gdjs.evtTools.linkedObjects._objectIsInList,
-          objectsLists,
-          false,
-          linkedObjects
-        );
+        )._getMapOfObjectsLinkedWith(obj);
+
+        let pickedSomething = false;
+        for (const contextObjectName in objectsLists.items) {
+          if (objectsLists.containsKey(contextObjectName)) {
+            const pickedObjects = objectsLists.items[contextObjectName];
+
+            if (pickedObjects.length === 0) {
+              continue;
+            }
+
+            // Find the object names in the scene
+            const temporaryObjectNames = gdjs.staticArray2(
+              gdjs.evtTools.linkedObjects.pickObjectsLinkedTo
+            );
+            temporaryObjectNames.length = 0;
+            if (isEventsFunction) {
+              // For functions, objects may be a flattened group
+              for (const pickedObject of pickedObjects) {
+                if (temporaryObjectNames.indexOf(pickedObject.getName()) < 0) {
+                  temporaryObjectNames.push(pickedObject.getName());
+                }
+              }
+            } else {
+              temporaryObjectNames.push(contextObjectName);
+            }
+
+            let objectCount = 0;
```

Comment:
Can you add a comment explaining what this objectCount and linkedObjectExists are doing?

Reply:
Something like `// Check if the picked object names are inside the linked objects` 
`// and count how many there are on the scene (to know later if we can avoid running an intersection between picked objects and the linked ones, and just replace the picked list with the linked object list`